tests/integration/cli/test_batch.py

```diff
@@ -58,7 +58,9 @@ def test_batch_run_grains_targeting(self):
 
         os_grain = os_grain.strip()
         cmd = self.run_salt(
-            '-C "G@os:{} and not localhost" -b 25% test.ping'.format(os_grain),
+            '-C "G@os:{} and not localhost" -b 25% test.ping'.format(
+                os_grain.replace(" ", "?")
```

Comment:
@bryceml I added this fix to the test that should make things work. We use fnmatch for globbing, and `?` matches any character. While this isn't a comprehensive fix, I think it's good enough. I tried a bunch of other methods:

- quotes: `G@os:"CentOS Stream"`
- backslash:  `G@os:CentOS\ Stream`
- brackets: `G@os:CentOS[ ]Stream`

I probably tried a few others that I'm forgetting now, but I wasn't able to figure out a way to get Salt to recognize any of those as valid, giving me "invalid compound target"

Unless someone else has a better solution, this was the best tradeoff:

- this test isn't actually testing how the OS grain was configured, only that grain targeting works
- for this test to allow a false positive (i.e. pass when it should have failed) we would literally have to have two OS grains configured that differ only by a space vs non-space like CentOS Stream and CentOSxStream.

It's reasonable enough, and the test *should* pass now.